src/elements/embed/embedComponents/EmbedRecommendations.tsx

```diff
@@ -35,7 +36,10 @@ export const EmbedRecommendation = ({
             embed code.{" "}
           </span>
           <ThemeProvider theme={buttonBrandAlt}>
-            <Button priority="secondary" onClick={() => convertTwitter(source)}>
+            <Button
+              priority="secondary"
+              onClick={() => convertTwitter(source as TwitterUrl)}
```

Comment:
Maybe I'm being a bit OTT, but if you write this with `source.startsWith("https://twitter.com/")` as a user-defined type guard you'd avoid the need for the cast.

It would be really nice if Typescript could infer this.

Reply:
As you say, a type guard probably the most elegant thing to do in the meantime – just done that.

Reply:
I worked out how to change the definition of startsWith to be a type guard, only to find that someone has already done it and it has already been rejected

https://github.com/microsoft/TypeScript/pull/46959/files
https://github.com/microsoft/TypeScript/issues/46958

Reply:
Well spotted – I'd never have thought to go back to the library typedef, good thinking. Can see the rationale.